### PR TITLE
[TT-7652] add disabled flag to middleware definition

### PR DESCRIPTION
**Pull request by jeffy-mathew**
## Related Issue
https://tyktech.atlassian.net/browse/TT-7652

### CODE REVIEW

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4664/merge
Commit:  
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3956268347)

File: gateway/server.go [hunk 1]
```diff
@@ -774,14 +774,18 @@ func (gw *Gateway) loadCustomMiddleware(spec *APISpec) ([]string, apidef.Middlew
 
 	// Load from the configuration
 	for _, mwObj := range spec.CustomMiddleware.Pre {
-		mwPaths = append(mwPaths, mwObj.Path)
-		mwPreFuncs = append(mwPreFuncs, mwObj)
-		mainLog.Debug("Loading custom PRE-PROCESSOR middleware: ", mwObj.Name)
+		if !mwObj.Disabled {
```
Comment (furkansenharputlu) [inline gateway/server.go:777]: can we use straight logic and use `continue`? and for others

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4664/merge
Commit: 245eec152bbeac463259018af06279b9623fea31 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3956590344)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4664/merge
Commit: 31a8a3eb643c8f8246ee303245b7fcba19313414 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3956630413)

**Review by titpetric — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4664/merge
Commit: 8c57173ab924b9f3e7e5be315b053505967d28b7 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3956727816)

**Review by furkansenharputlu — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4664/merge
Commit: 75809604d46637844d020f7bf27cf0217b6bdd26 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3957180620)

**Pull request merged**